File: HW1/simhw1.R

simulation <- function(g, seed, n = 1000000000) {
  y <- runif(n, 0, 1)
  sum <- sum(g(y))
  sum/n
}

g <- function(x) {
  return (1 - x)*x / (1 - 2*x + 2*x^2)^2
}

#system.time(result <- simulation(g, seed))
#Results:
#    user  system elapsed 
#  47.043  61.185 376.509 

simulation2 <- function(g, seed, n = 10^9) {
  rep <- 1
  if(n/10^7 > 1) {
    rep <- n/10^7
    n <- 10^7
  }
  
  total <- 0
  for(i in 1:rep) {
    y <- runif(n, 0, 1)
    total <- total + sum(g(y))
  }
  
  total/(n*rep)
}

results <- simulation2(g, seed = 10)

File: HW1/simhw1.c
//
//  main.c
//  HW 1
//
//  Created by Tyler Grimes on 1/14/16.
//  Copyright © 2016 Tyler Grimes. All rights reserved.
//

#include <stdio.h>
#include <stdlib.h>
#include <math.h>
#include <time.h>

double rand_unif();
double g(double x);
double simulation(unsigned int seed, int n);

int main(int argc, const char * argv[]) {
    
    unsigned int seed = 2;
    int n = 1000000000;
    
    double ans = simulation(seed, n);

    printf("Result = %lf\n", ans);
    
    return 0;
}

double rand_unif() {
    return rand()/(double)RAND_MAX;
}

double g(double x) {
    return ((1 - x)*(x)) / pow((1 - 2*x + 2*x*x), 2);
}

double simulation(unsigned int seed, int n) {
    srand(seed);
    
    double sum = 0;
    int i;
    for (i = 0; i < n; i++) {
        sum = sum + g(rand_unif());
    }
    
    return sum/n;   
}

